lib/services/presets.go

```diff
@@ -158,3 +159,49 @@ func NewPresetAuditorRole() types.Role {
 	role.SetLogins(types.Allow, []string{"no-login-" + uuid.New().String()})
 	return role
 }
+
+// DefaultAllowRules has the Allow rules that should be set as default when they were no explicitly defined.
+// This is used to update the current cluster roles when deploying a new resource.
+func DefaultAllowRules() map[string][]types.Rule {
+	return map[string][]types.Rule{
+		teleport.PresetEditorRoleName: {
+			types.NewRule(types.KindConnectionDiagnostic, RW()),
+		},
+	}
+}
+
+// AddDefaultAllowRules returns a copy of the received role with a new default allow rule.
```

Comment:
this says it returns a copy but it's actually modifying the original passed in rule

Reply:
Yeah
I guess we could even remove the return value or use it to indicate that no changes were made :thinking: 

I'll just remove the implementation details from the godoc for now